hibernate-core/src/main/java/org/hibernate/dialect/identity/PostgreSQLIdentityColumnSupport.java

```diff
@@ -30,18 +26,6 @@ public String getIdentitySelectString(String table, String column, int type) {
 
 	@Override
 	public String getIdentityColumnString(int type) {
-		switch (type) {
-			case Types.BIGINT:
-				return "bigserial not null";
-			case Types.INTEGER:
-				return "serial not null";
-			default:
-				throw new MappingException("illegal identity column type");
-		}
-	}
-
-	@Override
-	public boolean hasDataTypeInIdentityColumn() {
-		return false;
+		return "generated by default as identity";
```

Comment:
This should return the data type as well right? like 
```
case Types.BIGINT:
	return "bigint not null generated by default as identity";
case Types.INTEGER:
	return "integer not null generated by default as identity";
```
Or data type is already taken care by the caller of the function? If yes, now issues.

Reply:
Oh yes, thanks for clearing up the confusion.